engine/classes/ElggGroup.php

```diff
@@ -335,4 +335,75 @@ public function canComment($user_guid = 0, $default = null) {
 		}
 		return false;
 	}
+	
+	/**
+	 * Checks if a tool option is enabled
+	 *
+	 * @param string $option The option to check
+	 * @return bool
+	 * @since 3.0.0
+	 */
+	public function isToolEnabled($option) {
+		if (empty($option)) {
+			return false;
+		}
+		
+		$setting = $this->getPrivateSetting("tool:{$option}");
+		if ($setting !== null) {
+			return (bool) $setting;
+		}
+		
+		// check for old storage location in metadata
```

Comment:
Wouldn't it be better to add an upgrade script and get rid of the metadata related code completely?